docs/configuration.md

```diff
@@ -0,0 +1,457 @@
+>>Lets assume the following configuration:
+>>```js
+>>var config = {
+>>  playback: {
+>>    preferNative:  {
+>>      hls: true
+>>    }
+>>  }
+>>};
+>>```
+>> In case we're running in _Safari_ browser, the player will play native hls playback managed by the _Safari_ browser. But in case we're running in browser which hls playback is not supported natively, _Chrome_ for instance, the player will play hls using the `hls.js` library.
+>## 
+>>### config.playback.streamPriority
+>>##### Type: `Array<PKStreamPriorityObject`
+>>>##### Type `PKStreamPriorityObject`
+>>>```js
+>>>{
+>>>   engine: string,
+>>>   format: string
+>>>}
+>>>```
+>>##### Default: 
+>>```js
+>>[
+>>   {
+>>     engine: "html5",
+>>     format: "hls"
+>>   },
+>>   {
+>>     engine: "html5",
+>>     format: "dash"
+>>   },
+>>   {
+>>     engine: "html5",
+>>     format: "progressive"
+>>    }
+>>]
+>>```
+>>##### Description: Specifies the {engine, stream format} tuple priority of the player by ascending order.
+>>As soon as the player receives his sources, he will go over this array, and will try to play a source with a matched stream format from a matched engine. 
+>>For example, for the above priority configuration, the player will try to play hls stream using an html5 engine first. If an hls stream isn't received, he will continue to play dash stream using an html5 engine. If a dash stream isn't received, he will continue to play progressive stream using an html5 engine.
+
+### Workflow between `sources`, `preferNative` & `streamPriority`
+After we have gained knowledge of the different options available in the player configuration, let's see the results for what format will be played for the following configuration of the player:
+```js
+var config = {
+    sources: {
+        hls: [
+            {
+                mimetype: "application/x-mpegurl",
+                url: "https://wowzaec2demo.streamlock.net/vod-multitrack/_definst_/smil:ElephantsDream/ElephantsDream.smil/playlist.m3u8"
+            }
+        ]
+    },
+    playback: {
+        preferNative: {
+            hls: true,
+            dash: false
+        },
+        streamPriority: [
+            {
+                engine: "html5",
+                format: "dash"
+            },
+            {
+                engine: "html5",
+                format: "hls"
+            }
+        ]
+    }
+};
+```
+
+First, the player looks on the `streamPriority` array and see that *dash* stream is top priority. Second, he will check if he got any *dash* sources within the `sources` config object. Because no *dash* sources found there, the player will move to the next priority from the `streamPriority` array - *hls*. After checking again the sources config object, the player will choose to play *hls*. But is he play native *hls* or not? this depends on the `preferNative` value for *hls*. Because it sets to `true`, the player knows that if the browser supports native *hls* playback, it will be played natively.
```

Comment:
First, the player looks at the `streamPriority` array and see that the *dash* stream is top priority. Next, the player will will check whether it received any *dash* sources within the `sources` config object. Because there are no *dash* sources available, the player will move to the next priority from the `streamPriority` array, i.e., *hls*. After checking the sources config object again, the player will choose to play *hls*. However, will the player use the native *hls* or not? This depends on the `preferNative` value for *hls*. Because this value is set to `true`, the player knows that if the browser supports native *hls* playback, it will be played natively.